src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js

```diff
@@ -25,30 +29,50 @@ type Props = {
     translations?: Translations,
 };
 
-const ActiveState = ({
-    currentUser,
-    items,
-    onCommentDelete,
-    onTaskDelete,
-    onTaskEdit,
-    onTaskAssignmentUpdate,
-    onVersionInfo,
-    translations,
-    getAvatarUrl,
-    getUserProfileUrl,
-    getMentionWithQuery,
-    mentionSelectorContacts,
-}: Props): React.Node => (
-    <ul className="bcs-activity-feed-active-state">
-        {items.map((item: any) => {
-            const { type, id, versions, permissions } = item;
+class ActiveState extends React.Component<Props> {
+    cache: any;
+
+    list: any;
+
+    constructor() {
+        super();
+        this.cache = new CellMeasurerCache({
+            defaultHeight: 81,
+            fixedWidth: true,
+        });
+    }
+
+    componentDidUpdate() {
```

Comment:
We might be able to be more efficient here by only updating rows that changed.

Reply:
Else it'd re-render on any prop change.